From 67dd879fbe62bc28d3bbc48b88c3842e2cb827e8 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Fri, 09 Sep 2011 07:51:54 +0000
Subject: gsm: Never mark as handled the Disconnected signal from dbus

Disconnected is emitted when dbus is quitting. And all the dbus filters
might be interested in this signal, so never mark it as handled when we
look at it.
---
diff --git a/gnome-session/gsm-consolekit.c b/gnome-session/gsm-consolekit.c
index f53993f..66bebd1 100644
--- a/gnome-session/gsm-consolekit.c
+++ b/gnome-session/gsm-consolekit.c
@@ -152,7 +152,8 @@ gsm_consolekit_dbus_filter (DBusConnection *connection,
                                     DBUS_INTERFACE_LOCAL, "Disconnected") &&
             strcmp (dbus_message_get_path (message), DBUS_PATH_LOCAL) == 0) {
                 gsm_consolekit_free_dbus (manager);
-                return DBUS_HANDLER_RESULT_HANDLED;
+                /* let other filters get this disconnected signal, so that they
+                 * can handle it too */
         }
 
         return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
diff --git a/gnome-session/gsm-shell.c b/gnome-session/gsm-shell.c
index 0b26f09..9785160 100644
--- a/gnome-session/gsm-shell.c
+++ b/gnome-session/gsm-shell.c
@@ -228,7 +228,8 @@ gsm_shell_bus_filter (DBusConnection *connection,
                                     DBUS_INTERFACE_LOCAL, "Disconnected") &&
             strcmp (dbus_message_get_path (message), DBUS_PATH_LOCAL) == 0) {
                 gsm_shell_disconnect_from_bus (shell);
-                return DBUS_HANDLER_RESULT_HANDLED;
+                /* let other filters get this disconnected signal, so that they
+                 * can handle it too */
         }
 
         return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
--
cgit v0.9.0.2
